Arduino/MiP_ProMini_Pack_Library/src/mip.cpp

```diff
@@ -1814,7 +1816,90 @@ int8_t MiP::rawGetGameMode(MiPGameMode& mode)
 
     // Restart the game mode now that we have successfully retrieved it.
     rawSetGameMode(mode);
-    
+
+    return MIP_ERROR_NONE;
+}
+
+
+
+void MiP::setUserData(uint8_t address, uint8_t userData)
+{
+    verifiedSetUserData(address, userData);
+}
+
+int8_t MiP::getUserData(uint8_t address, uint8_t& userData)
```

Comment:
If you use existing APIs like readRadar(), readGestureEvent(), readVolume(), etc which return simple integers or enumerations as an example then it would probably be better to get rid of the userData parameter and just return the read value instead. I don't return errors from most of the public functions since the average Arduino user is just going to assume that their calls succeeded and won't write error handling code. That is why I added all of the read after write and retries down in the library so that I can isolate the average Arduino developer from as much error handling as possible.

I would continue to have rawGetUserData() to use the userData parameter and return an error code since that is consistent with most of the existing raw APIs.